src/ui/public/chrome/directives/kbn_chrome.js

```diff
@@ -43,6 +43,29 @@ export default function (chrome, internals) {
         // and some local values
         chrome.httpActive = $http.pendingRequests;
         $scope.notifList = require('ui/notify')._notifs;
+
+        // App switcher functionality.
+        function updateAppSwitcher(isOpen) {
+          $scope.appSwitcherButton = {
+            classes: isOpen ? 'app-switcher-link--close' : undefined,
+            title: isOpen ? 'Collapse' : 'Expand',
+            tooltip: isOpen ? 'Collapse side bar' : 'Expand side bar',
+          };
+        }
+
+        $scope.isAppSwitcherOpen = appSwitcherState.isOpen();
+        updateAppSwitcher($scope.isAppSwitcherOpen);
+
+        $scope.toggleAppSwitcher = event => {
+          event.preventDefault();
+          $scope.isAppSwitcherOpen = !$scope.isAppSwitcherOpen;
+          appSwitcherState.setOpen($scope.isAppSwitcherOpen);
+          updateAppSwitcher($scope.isAppSwitcherOpen);
```

Comment:
If `appSwitcherState.setOpen` gets called outside of this `toggleAppSwitcher` function, `$scope.isAppSwitcherOpon` won't get updated. I'm always wary when I see scope variables and services that represent the same thing, but aren't linked via some kind of watcher that guarantees (as much as possible) they'll stay in sync.

Perhaps the `appSwitcherState` service should emit an event on change, and this directive should listen for it. The listener for that event would update `$scope.isAppSwitcherOpen` and do the work of `updateAppSwitcher`. `toggleAppSwitcher` would simply have to call `appSwitcherState.setOpen`. This would be a bit more fluxy.